src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -214,10 +214,15 @@ namespace xclcpuemhal2 {
     }
     if (buf_size < new_size)
     {
-      void *temp = buf;
-      buf = (void*) realloc(buf,new_size);
-      if (!buf) // prevent leak of original buf
-        free(temp);
+      void *result = realloc(buf, new_size);
+      // If realloc was unsuccessful, then give up and deallocate.
+      if (result == nullptr)
```

Comment:
```suggestion
      if (!result)
```
Checking against nullptr is just silly and C++ or C style at all.

Reply:
AFAIK `nullptr` was introduced in C++11 to distinguish it from arithmetic zero (as in `NULL`). This avoids problems when we have two overloaded functions where one function has an `int` as an argument and the other a pointer as an argument. I thought it is better to be more clear in the if statement. But [I read that `nullptr` evaluates to `false` in a boolean context](https://stackoverflow.com/a/57527724), so your solution is shorter. Thanks!

Can you elaborate on why is this silly? I am just curious.

Reply:
Sure it's not really a big deal.   Absolutely nullptr in place of NULL of course.

But checking against nullptr is redundant and doesn't make the code more readable, IMO.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-if

Reply:
After reading the guideline your comment makes more sense, thanks @stsoe